### PR TITLE
Dumb component widget DESKTOP-842

### PR DESCRIPTION
**Pull request by awendland**
Expand on the dumb components for the menubar (AKA widget) to include both the public and private modes, as well as the open popup menu state. This will improve the ability for Viss Diff to track changes in the menu bar.

screen shot 2016-06-02 at 5 49 52 pm
screen shot 2016-06-02 at 5 47 47 pm
screen shot 2016-06-02 at 5 47 56 pm

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Added: **Menubar-Public_Truncated**  
   Menubar-Public_Truncated rendered
 * Added: **Menubar-Public_Normal**  
   Menubar-Public_Normal rendered
 * Added: **Menubar-Public_Menu_Showing**  
   Menubar-Public_Menu_Showing rendered
 * Added: **Menubar-Private_Truncated**  
   Menubar-Private_Truncated rendered
 * Added: **Menubar-Private_Normal**  
   Menubar-Private_Normal rendered
 * Added: **Menubar-Private_Menu_Showing**  
   Menubar-Private_Menu_Showing rendered
 * Removed: **Menubar-Truncated** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464921109522-05ed1f4768b3-41b8e67fe8d8/Menubar-Truncated-05ed1f4768b3.png)
 * Removed: **Menubar-Normal** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1464921109522-05ed1f4768b3-41b8e67fe8d8/Menubar-Normal-05ed1f4768b3.png)

File: shared/menubar/index.render.desktop.js [hunk 1]
```diff
@@ -88,19 +94,19 @@ class Render extends Component<void, Props, State> {
...
-    const styles = this.state.showingPublic ? stylesPublic : stylesPrivate
+    const styles = this.state.showingPrivate ? stylesPrivate : stylesPublic
 
     const mergedProps = {
       ...this.props,
       smallMode: true,
       private: newPrivate,
       public: newPublic,
-      onSwitchTab: showingPublic => this.setState({showingPublic})
+      onSwitchTab: showingPublic => this.setState({showingPrivate: !showingPublic})
```
Comment (chromakode) [inline shared/menubar/index.render.desktop.js]: I think you should rename the `showingPublic` param here.

**Comment by chromakode** [comment]
In general props that set initial state params tend to be an anti-pattern -- they often indicate state that should be moved up to the parent component and controlled via `props`. In this case, I think it's probably fine to do the simple thing here and refactor when we need more control

:+1: % param name nit

File: shared/menubar/index.render.desktop.js (hunk 1 above)
Comment (chromakode) [inline shared/menubar/index.render.desktop.js]: Er, nevermind! The broken syntax highlighting made it hard to spot the `!` here.

**Comment by keybase-ci-visdiff** [comment]
:mag_right: These commits introduced some visual changes:
 * Added: **Menubar-Public_Truncated**  
   Menubar-Public_Truncated rendered
 * Added: **Menubar-Public_Normal**  
   Menubar-Public_Normal rendered
 * Added: **Menubar-Public_Menu_Showing**  
   Menubar-Public_Menu_Showing rendered
 * Added: **Menubar-Private_Truncated**  
   Menubar-Private_Truncated rendered
 * Added: **Menubar-Private_Normal**  
   Menubar-Private_Normal rendered
 * Added: **Menubar-Private_Menu_Showing**  
   Menubar-Private_Menu_Showing rendered
 * Added: **Folders_TLF-Normal_Public**  
   Folders_TLF-Normal_Public rendered
 * Added: **Folders_TLF-Normal_Private** [(view)](http://keybase-app-visdiff.s3.amazonaws.com/1465424224894-f856e4286e76-dc6d8f0b6ebb/Folders_TLF-Normal_Private-dc6d8f0b6ebb.png)
 * Removed: **Menubar-Truncated** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1465424224894-f856e4286e76-dc6d8f0b6ebb/Menubar-Truncated-f856e4286e76.png)
 * Removed: **Menubar-Normal** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1465424224894-f856e4286e76-dc6d8f0b6ebb/Menubar-Normal-f856e4286e76.png)
 * Removed: **Folders_TLF-Normal** [(view original)](http://keybase-app-visdiff.s3.amazonaws.com/1465424224894-f856e4286e76-dc6d8f0b6ebb/Folders_TLF-Normal-f856e4286e76.png)

File: shared/folders/render.js.flow [hunk 2]
```diff
@@ -11,8 +11,9 @@ export type Props = {
   public: ListProps,
   onSwitchTab?: (showingPublic: boolean) => void,
   listStyle?: any,
-  smallMode: boolean,
```
Comment (chromakode) [inline shared/folders/render.js.flow]: :sheep: I think this snuck in from another commit in this set

**Comment by chromakode** [comment]
:+1:

**Pull request merged**